DataFormats/Provenance/interface/Provenance.h

```diff
@@ -38,48 +35,47 @@ namespace edm {
 
     Provenance(std::shared_ptr<BranchDescription const> const& p, ProductID const& pid);
 
-    BranchDescription const& product() const {return *branchDescription_;}
+    Provenance(StableProvenance const&);
 
-    BranchDescription const& branchDescription() const {return *branchDescription_;}
-    BranchDescription const& constBranchDescription() const {return *branchDescription_;}
-    std::shared_ptr<BranchDescription const> const& constBranchDescriptionPtr() const {return branchDescription_;}
+    StableProvenance const& stable() const {return stableProvenance_;}
+    StableProvenance& stable() {return stableProvenance_;}
+
+    BranchDescription const& product() const {return branchDescription();}
+    BranchDescription const& branchDescription() const {return stable().branchDescription();}
+    BranchDescription const& constBranchDescription() const {return branchDescription();}
+    std::shared_ptr<BranchDescription const> const& constBranchDescriptionPtr() const {return stable().constBranchDescriptionPtr();}
 
```

Comment:
We really need to go through the code and remove calls to `product` and `constBranchDescription` and just use `branchDescrption`. However, that would be for another pull request.